test/processor/StructureDefinitionProcessor.test.ts

```diff
@@ -0,0 +1,257 @@
+import path from 'path';
+import fs from 'fs-extra';
+import { fhirdefs, fshtypes } from 'fsh-sushi';
+import {
+  StructureDefinitionProcessor,
+  ProcessableElementDefinition,
+  ProcessableStructureDefinition
+} from '../../src/processor';
+import {
+  ExportableProfile,
+  ExportableExtension,
+  ExportableCardRule,
+  ExportableAssignmentRule,
+  ExportableCaretValueRule,
+  ExportableObeysRule,
+  ExportableContainsRule,
+  ExportableInvariant
+} from '../../src/exportable';
+import '../helpers/loggerSpy'; // suppresses console logging
+
+describe('StructureDefinitionProcessor', () => {
+  let defs: fhirdefs.FHIRDefinitions;
+
+  beforeAll(() => {
+    defs = new fhirdefs.FHIRDefinitions();
+    fhirdefs.loadFromPath(path.join(__dirname, '..', 'utils', 'testdefs'), 'testPackage', defs);
+  });
+
+  describe('#process', () => {
+    it('should convert the simplest Profile', () => {
+      const input = JSON.parse(
+        fs.readFileSync(path.join(__dirname, 'fixtures', 'simple-profile.json'), 'utf-8')
+      );
+      const expectedProfile = new ExportableProfile('SimpleProfile');
+      const result = StructureDefinitionProcessor.process(input, defs);
+
+      expect(result).toContainEqual<ExportableProfile>(expectedProfile);
+    });
+
+    it('should convert the simplest Extension', () => {
+      const input = JSON.parse(
+        fs.readFileSync(path.join(__dirname, 'fixtures', 'simple-extension.json'), 'utf-8')
+      );
+      const expectedExtension = new ExportableExtension('SimpleExtension');
+      const result = StructureDefinitionProcessor.process(input, defs);
+
+      expect(result).toContainEqual<ExportableExtension>(expectedExtension);
+    });
+
+    it('should convert a profile that has a baseDefinition', () => {
+      // the baseDefinition field defines a Profile's Parent
+      const input = JSON.parse(
+        fs.readFileSync(path.join(__dirname, 'fixtures', 'big-profile.json'), 'utf-8')
+      );
+      const [profile] = StructureDefinitionProcessor.process(input, defs);
+
+      expect(profile).toBeInstanceOf(ExportableProfile);
+      expect(profile.name).toBe('BigProfile');
+      expect(profile.parent).toBe('https://demo.org/StructureDefinition/SmallProfile');
+    });
+
+    it('should convert a profile with an invariant', () => {
```

Comment:
Is there a reason you took off the comparison of the `invariants` list to the `ExportableInvariant` that was in the old ProfileProcessor tests?

Reply:
The correctness of the invariant is part of testing `InvariantExtractor`. Here, we can assume that the extractor will produce correct results, and we just want to test that we got an invariant.

Reply:
Ah! That makes sense! Thanks for the explanation!